### DISCUSSION TITLE
how to integrate Tag Manager

### DISCUSSION THREAD

**Discussion by mukundkadu0** [arthelokyo/astrowind#216] [Q&A] [answered]
Hello fellow forum members,

I hope you're all doing well. I have a question regarding the integration of Google Tag Manager (GTM), and I was wondering if someone could kindly provide me with some guidance.

Could someone please help me by elaborating on the process of integrating Google Tag Manager into a website theme? What are the steps I need to follow? How do I set up tags and triggers within GTM? Any tips or best practices would be greatly appreciated.

Thank you in advance for your assistance! Your expertise will be invaluable in helping me understand and successfully implement Google Tag Manager.

**Comment by prototypa** [comment]
Hello @mikkivirus-dev

I don't know the configuration details of the Google Tag Manager because I have never used it, although I know that it allows you to inject scripts into the pages.  There is a way, for example, to inject Google Analytics using Google Tag Manager.

I can suggest you in which file of the template you can place the code provided by Google Tag Manager, from that place it will fire on all the pages of the website: https://github.com/onwidget/astrowind/blob/main/src/layouts/Layout.astro

Another user can help you with the other details you need.

Thank you

**Reply by mukundkadu0** [reply]
Saved My Day Thanks

**Comment by thomHayner** [comment, marked as answer]
@prototypa I also wanted to add Google Tag Manager to my project, I followed the same pattern as your [@astrolib/analytics](https://www.npmjs.com/package/@astrolib/analytics?activeTab=dependencies) package for Google Analytics.  It's just one extra/alternate script (Google will disable your site's GA4 tags and force you to go through GTM if you create a GTM property for your website).

You are welcome to it, anyone else may use it as well.

Tested and working on localhost in Google Tags Debug Mode and on my website deployed to production.

Semi-packaged code is [here](https://github.com/ScaleForceAgency/scaleforce-agency-landing/tree/blogTemplates/src/integrations/google-tag-manager).
Other necessary updates to configBuilder.ts and config-schema.json are [here](https://github.com/ScaleForceAgency/scaleforce-agency-landing/commit/d62b6f12e87f055f07d2bac82111416bb45b0af1).
As well as config.yaml updates [here](https://github.com/ScaleForceAgency/scaleforce-agency-landing/commit/3037823be6acad0559e840ff0e2ffa534f6cc060).
Then add two new components to src/components/common, [BodyTag here](https://github.com/ScaleForceAgency/scaleforce-agency-landing/blob/main/src/components/common/TagsBody.astro), [HeadTag here](https://github.com/ScaleForceAgency/scaleforce-agency-landing/blob/main/src/components/common/TagsHead.astro).
Then insert the BodyTag and HeadTag to Layout.astro as seen [here](https://github.com/ScaleForceAgency/scaleforce-agency-landing/commit/869ca60020def1ff0a3af566d7eaa53a96c1c671).

**Reply by porobertdev** [reply]
Single commit as patch (private repo, sorry). Credits to thom.

1. Change `.txt` to `.patch`.
2. `git am add-gtm-astro.patch` (more [here](https://stackoverflow.com/a/2250170)).


[add-gtm-astro.txt](https://github.com/user-attachments/files/21551537/add-gtm-astro.txt)
